Turn off scalar_checks for torch.clamp.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30502 [BC-BREAKING] change index_select scalar_check to retain dimensionality of input.
* #30438 Turn off scalar_checks for SpatialDepthwiseConvolution and SpatialConvolutionMM.
* #30437 [WIP] Remove scalar_check from topk, move it to the THC implementation.
* #30436 Turn off scalar_checks for _th_reciprocal.
* **#30435 Turn off scalar_checks for torch.clamp.**
* #30434 Turn off scalar_checks for exp, cos, cosh, tan, atan, tanh, erf, erfc.

The underlying THC implementations are correct.

Differential Revision: [D18699089](https://our.internmc.facebook.com/intern/diff/D18699089)

gchanan committed:
Turn off scalar_checks for torch.clamp.

The underlying THC implementations are correct.

[ghstack-poisoned]

gchanan committed:
Update on "Turn off scalar_checks for torch.clamp."


The underlying THC implementations are correct.

[ghstack-poisoned]

gchanan committed:
Update on "Turn off scalar_checks for torch.clamp."


The underlying THC implementations are correct.

[ghstack-poisoned]

gchanan committed:
Update on "Turn off scalar_checks for torch.clamp."


The underlying THC implementations are correct.

Differential Revision: [D18699089](https://our.internmc.facebook.com/intern/diff/D18699089)

[ghstack-poisoned]

gchanan committed:
Update on "Turn off scalar_checks for torch.clamp."


The underlying THC implementations are correct.

Differential Revision: [D18699089](https://our.internmc.facebook.com/intern/diff/D18699089)

[ghstack-poisoned]

albanD commented on test/test_torch.py:
```diff
@@ -6109,6 +6109,14 @@ def test_scalar_check(self, device):
         self.assertEqual((1,), torch.erfc(one_d).shape)
 
+        # clamp
+        self.assertEqual((), torch.clamp(zero_d, min=0, max=1).shape)
```
nit: the fact that these are `int`s is not a problem?

gchanan replied:
nope, the function handles them correctly.

gchanan committed:
Update on "Turn off scalar_checks for torch.clamp."


The underlying THC implementations are correct.

Differential Revision: [D18699089](https://our.internmc.facebook.com/intern/diff/D18699089)

[ghstack-poisoned]

gchanan committed:
Update on "Turn off scalar_checks for torch.clamp."


The underlying THC implementations are correct.

Differential Revision: [D18699089](https://our.internmc.facebook.com/intern/diff/D18699089)

[ghstack-poisoned]

gchanan committed:
Update on "Turn off scalar_checks for torch.clamp."


The underlying THC implementations are correct.

Differential Revision: [D18699089](https://our.internmc.facebook.com/intern/diff/D18699089)

[ghstack-poisoned]

gchanan committed:
Update on "Turn off scalar_checks for torch.clamp."


The underlying THC implementations are correct.

Differential Revision: [D18699089](https://our.internmc.facebook.com/intern/diff/D18699089)

[ghstack-poisoned]
